code/drasil-printers/Language/Drasil/DOT/Print.hs

```diff
@@ -121,20 +121,20 @@ mkOutputAllvsAll gi = do
 -------------
 
 -- | General output function for making a traceability graph. Takes in the graph information, title, direction function, label functions, label prefixes, and colour functions.
-mkOutput :: GraphInfo -> String -> (GraphInfo -> [(String, [String])]) -> [GraphInfo -> ([String], String)] -> [GraphInfo -> Colour] -> IO ()
-mkOutput gi title getDirections getLabels getColours = do
-    handle <- openFile (title ++ ".dot") WriteMode
-    hPutStrLn handle $ "digraph " ++ title ++ " {"
+mkOutput :: GraphInfo -> String -> (GraphInfo -> [(String, [String])]) -> [GraphInfo -> (([String], [String]), String)] -> [GraphInfo -> Colour] -> IO ()
+mkOutput gi ttl getDirections getLabels getColours = do
+    handle <- openFile (ttl ++ ".dot") WriteMode
+    hPutStrLn handle $ "digraph " ++ ttl ++ " {"
     let labels = filterAndGI gi getLabels
         colours = map ($ gi) getColours
 
-        allLabels = zipWith (\x y -> (fst x, snd x, y)) labels colours
+        allLabels = zipWith (\x y -> (fst $ fst x, snd $ fst x, snd x, y)) labels colours
```

Comment:
use pattern-matching for x and y in the function so that you don't need to have this string of `fst` and `snd`.

Plus this is likely an abuse of tuples. Never ever use a 4-tuple, always make that a record. 3-tuples should be rather rare too. I don't think there's a good reason to use a tuple here, so I'd create a record.